### PR TITLE
ES report PR feedback

### PR DESCRIPTION
**Pull request by benrudolph**
@snopoke this is feedback from previous report PRs. main thing is switch the application status report to use one query instead of N. :dolphin: or :hospital: 

cc: @gcapalbo

### CODE REVIEW

File: corehq/apps/es/aggregations.py [hunk 1]
```diff
@@ -111,6 +111,28 @@ def counts_by_bucket(self):
         return {b['key']: b['doc_count'] for b in self.normalized_buckets}
 
 
+class TopHitsResult(AggregationResult):
```
Comment (benrudolph) [inline corehq/apps/es/aggregations.py]: this is pretty similar to ESQuerySet, but seemed messy to try and be completely DRY. not that much copied code

File: corehq/apps/es/aggregations.py [hunk 2]
```diff
@@ -208,6 +230,31 @@ class ExtendedStatsAggregation(StatsAggregation):
     result_class = ExtendedStatsResult
 
 
+class TopHitsAggregation(Aggregation):
+    """
+    Stats aggregation that computes a stats aggregation by field
```
Comment (snopoke) [inline corehq/apps/es/aggregations.py]: Doctoring incorrect

File: corehq/apps/es/aggregations.py [hunk 3]
```diff
@@ -208,6 +230,31 @@ class ExtendedStatsAggregation(StatsAggregation):
...
+            "Names must be valid python variable names, was {}".format(name)
+        self.name = name
+        self.body = {
+            "sort": [{
+                field: {
+                    "order": 'asc' if is_ascending else 'desc'
+                },
+            }],
+            'size': size,
+        }
```
Comment (snopoke) [inline corehq/apps/es/aggregations.py]: Would be nice to also be able to specify the source mapping to be fully compatible with the spec

**Comment by snopoke** [comment]
Minor feedback but looks great

File: corehq/apps/es/aggregations.py (hunk 2 above)
Comment (benrudolph) [inline corehq/apps/es/aggregations.py]: doh: https://github.com/dimagi/commcare-hq/commit/4a909f2d67438c3cab14c64d0e019a74536cba93

File: corehq/apps/es/aggregations.py (hunk 3 above)
Comment (benrudolph) [inline corehq/apps/es/aggregations.py]: :+1: https://github.com/dimagi/commcare-hq/commit/4a909f2d67438c3cab14c64d0e019a74536cba93 also checked to make the TopHitsResult still works. still didn't implement the ability to sort by multiple fields, but think we can do that if we ever have the need

**Pull request merged**